### PR TITLE
touch: add --recursive flag - issue #5301

### PR DESCRIPTION
**Pull request by aeromaxx**
#### Checklist

- [ ] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [x] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [ ] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ ] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by aeromaxx** [comment]
@ncw Do you have any recommendations as to what to do with this next?

File: cmd/touch/touch.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ func init() {
 	flags.BoolVarP(cmdFlags, &notCreateNewFile, "no-create", "C", false, "Do not create the file if it does not exist.")
 	flags.StringVarP(cmdFlags, &timeAsArgument, "timestamp", "t", "", "Use specified time instead of the current time of day.")
 	flags.BoolVarP(cmdFlags, &localTime, "localtime", "", false, "Use localtime for timestamp, not UTC.")
+	flags.BoolVarP(cmdFlags, &recurse, "recursive", "R", false, "Recurse into the listing.")
```
Comment (albertony) [inline cmd/touch/touch.go:34]: You need to declare variable `recurse` above, same place as `localTime`.

File: cmd/touch/touch.go [hunk 2]
```diff
@@ -101,3 +102,33 @@ func Touch(ctx context.Context, fsrc fs.Fs, srcFileName string) (err error) {
...
+	if timeAsArgument != "" {
+		layout := defaultLayout
+		if len(timeAsArgument) == len(layoutDateWithTime) {
+			layout = layoutDateWithTime
+		} else if len(timeAsArgument) > len(layoutDateWithTime) {
+			layout = layoutDateWithTimeNano
+		}
+		var timeAtrFromFlags time.Time
+		if localTime {
+			timeAtrFromFlags, err = time.ParseInLocation(layout, timeAsArgument, time.Local)
```
Comment (albertony) [inline cmd/touch/touch.go:118]: You must also declare `err`, e.g. `var err error` just below `var timeAtrFromFlags time.Time` above the if.

File: cmd/touch/touch.go [hunk 3]
```diff
@@ -101,3 +102,33 @@ func Touch(ctx context.Context, fsrc fs.Fs, srcFileName string) (err error) {
...
+		} else {
+			timeAtrFromFlags, err = time.Parse(layout, timeAsArgument)
+		}
+		if err != nil {
+			return errors.Wrap(err, "failed to parse date/time argument")
+		}
+		timeAtr = timeAtrFromFlags
+	}
+	return ListFn(ctx, f, func(o fs.Object) {
```
Comment (albertony) [inline cmd/touch/touch.go:127]: Maybe this needs to be `operations.ListFn...`, and with an `"github.com/rclone/rclone/fs/operations"` entry in import up in the beginning of the file.

File: cmd/touch/touch.go (hunk 1 above)
Comment (aeromaxx) [reply cmd/touch/touch.go:34]: @albertony Ah thank you for that, I have done that now.

File: cmd/touch/touch.go (hunk 2 above)
Comment (aeromaxx) [reply cmd/touch/touch.go:118]: @albertony Ah thank you for that, I have done that now.

File: cmd/touch/touch.go (hunk 3 above)
Comment (aeromaxx) [reply cmd/touch/touch.go:127]: @ncw What would you advise to do here? Just you gave me the code for the ListFn function.  https://forum.rclone.org/t/using-rclone-touch-recursively/23975/13

**Comment by aeromaxx** [comment]
@albertony @ncw This issue has a new pull request here - https://github.com/rclone/rclone/pull/5330

**Pull request closed without merging**